### ISSUE TITLE
Add support for the dSYM for a prebuilt binary.

### ISSUE THREAD

**Issue by alloy** [CocoaPods/CocoaPods#1698]
When a prebuilt binary has been stripped of its debug symbols, adding this will still allow for a better debugging experience during development.

From #1696:

> LLDB supports remapping of dSYM source paths via embedded UUID property lists: http://lldb.llvm.org/symbols.html

**Comment by kylef** [comment]
@segiddins Yes, I think a framework should behave the same way.

```ruby
s.vendored_framework = { :name => 'Stencil.framework', :dSYM => 'Stencil.framework.dSYM' }
```

```shell
├── Stencil.framework
│   ├── Headers -> Versions/Current/Headers
│   ├── Modules -> Versions/Current/Modules
│   ├── Resources -> Versions/Current/Resources
│   ├── Stencil -> Versions/Current/Stencil
│   └── Versions
│       ├── A
│       │   ├── Headers
│       │   │   ├── Stencil-Swift.h
│       │   │   └── Stencil.h
│       │   ├── Modules
│       │   │   ├── Stencil.swiftmodule
│       │   │   │   ├── x86_64.swiftdoc
│       │   │   │   └── x86_64.swiftmodule
│       │   │   └── module.modulemap
│       │   ├── Resources
│       │   │   └── Info.plist
│       │   └── Stencil
│       └── Current -> A
└── Stencil.framework.dSYM
    └── Contents
        ├── Info.plist
        └── Resources
            └── DWARF
                └── Stencil

15 directories, 10 files
```

**Comment by segiddins** [comment]
@kylef is there no place in the `.framework` itself to put a dSYM?

**Comment by kylef** [comment]
I'm not sure, but this is how Xcode generates it, separately.

**Comment by ulhas** [comment]
+1 for this issue. Would be awesome for CocoaPods to have support for dsyms for a framework.

**Comment by segiddins** [comment]
@ulhas the quickest way to get a feature like this into CocoaPods would be to submit a PR for it :rocket:

**Comment by AlexeyDemedetskiy** [comment]
Basically dSYM and .bcsymbolmap support is a copy 3 files to Products directory. Is there any hook where this files can be specified? Like resources. [Script example](http://developers.atinternet-solutions.com/wp-content/uploads/2015/06/copy_files_phase.png)

**Comment by dnkoutso** [comment]
I know this is an old issue but we just hit this case and worth adding support. The proposed API seems good.

**Comment by piotr-tobolski** [comment]
Can this be closed? #6536 was merged

I hope this will be released soon 👍

**Comment by dnkoutso** [comment]
Not sure if it should. The original issue was discussing adding a DSL option. Let's wait and see how the current implementation that assumes the same name goes.

**Comment by falcon283** [comment]
I'm really looking forward to have this feature.
I'm planning to use vendored SDK for the project I'm working on, and the ability to just link them instead of instead of building them every time sounds awesome but I really need to wrap up all the dSYM into the final application dSYM in order to easily track down crashes in my App and frameworks too.

Is this intended for this purpose? Is there any planned release date for this?

**Comment by dnkoutso** [comment]
If you use bundler you can lock CocoaPods to a specific SHA. 

This should ship this week or next with 1.3.0.beta.1.

**Comment by colinsf** [comment]
@dnkoutso is there usage documentation for #6536 ?  maybe an example .podspec?

**Comment by dnkoutso** [comment]
No, there is no DSL added for it. If you attach a dSYM next to the .framework then it will automatically be detected and processed by CocoaPods.

**Comment by dnkoutso** [comment]
For anyone who wants to follow progress here, CocoaPods now supports vendored `dSYM` support but only if the name of the `dSYM` matches that of the framework.

This issue will remain open to extend the DSL to allow pod authors to specify a specific/different name of a `dSYM` file.

**Comment by rostopira** [comment]
@dnkoutso could you please make it clear, how it should match?
If I have `WebRTC.framework`, should I place a `WebRTC.dSYM` or `WebRTC.framework.dSYM` in same folder?
Thaks

**Comment by dnkoutso** [comment]
@rostopira mist be `WebRTC.framework.dSYM`
